api/app/Http/Controllers/OccurrenceController.php

```diff
@@ -718,41 +728,55 @@ public function oneOccurrenceReharvest($id, Request $request){
 	}
 
 	//Helper functions
-	private function isAuthorized($user, $collid){
-		foreach($user['roles'] as $roles){
-			if($roles['role'] == 'SuperAdmin') return true;
-			elseif($roles['role'] == 'CollAdmin' && $roles['tablePK'] == $collid) return true;
-			elseif($roles['role'] == 'CollEditor' && $roles['tablePK'] == $collid) return true;
+	protected function getOccid($id){
+		if(!is_numeric($id)){
+			$occid = Occurrence::where('occurrenceID', $id)->orWhere('recordID', $id)->value('occid');
+			if(is_numeric($occid)) $id = $occid;
+		}
+		return $id;
+	}
+
+	private function isAuthorized($user, $collid) {
+		foreach ($user['roles'] as $roles) {
+			if ($roles['role'] == 'SuperAdmin') return true;
+			elseif ($roles['role'] == 'CollAdmin' && $roles['tablePK'] == $collid) return true;
+			elseif ($roles['role'] == 'CollEditor' && $roles['tablePK'] == $collid) return true;
 		}
 		return false;
 	}
 
-	protected function getAPIResponce($url, $asyc = false){
+	protected function getAPIResponce($url, $asyc = false) {
 		$resJson = false;
 		$ch = curl_init();
 		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
 		curl_setopt($ch, CURLOPT_URL, $url);
 		//curl_setopt($ch, CURLOPT_HTTPGET, true);
 		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
 		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
-		if($asyc) curl_setopt($ch, CURLOPT_TIMEOUT_MS, 500);
+		if ($asyc) curl_setopt($ch, CURLOPT_TIMEOUT_MS, 500);
 		$resJson = curl_exec($ch);
-		if(!$resJson){
-			$this->errorMessage = 'FATAL CURL ERROR: '.curl_error($ch).' (#'.curl_errno($ch).')';
+		if (!$resJson) {
+			$this->errorMessage = 'FATAL CURL ERROR: ' . curl_error($ch) . ' (#' . curl_errno($ch) . ')';
 			return false;
 			//$header = curl_getinfo($ch);
 		}
 		curl_close($ch);
-		return json_decode($resJson,true);
+		return json_decode($resJson, true);
 	}
 
-	private function getOccurrence($id){
+	private function getOccurrence($id) {
 		$decodedId = urldecode($id);
 		$occurrence = null;
-		if(is_numeric($decodedId)){
-			$occurrence = Occurrence::find($decodedId);
-		} else{
-			$occurrence = Occurrence::where('recordID', (string)$decodedId)
+		if (is_numeric($decodedId)) {
+			$occurrence = DB::table('omoccurrences as o')
+				->select('o.*', 't.author', 't.sciName as trueSciName')
```

Comment:
Getting author and sciName from taxa table in case omoccurrences version doesn't exist or is old.